components/provisioner/internal/gardener/auditlogging.go

```diff
@@ -34,6 +36,98 @@ func (a *auditLogConfigurator) CanEnableAuditLogsForShoot(seedName string) bool
 	return seedName != "" && a.auditLogTenantConfigPath != ""
 }
 
+func findSecret(shoot *gardener_types.Shoot) *gardener_types.NamedResourceReference {
+	// shoot.GetAudit
+	for i, e := range shoot.Spec.Resources {
+		if e.Name == "auditlog-credentials" {
+			return &shoot.Spec.Resources[i]
+		}
+	}
+
+	return nil
+}
+
+func configureSecret(shoot *gardener_types.Shoot, config AuditLogConfig) (changed bool) {
+	changed = false
+
+	sec := findSecret(shoot)
+	if sec != nil {
+		if sec.Name == "auditlog-credentials" &&
+		sec.ResourceRef.APIVersion == "v1" &&
+		sec.ResourceRef.Kind == "Secret" &&
+		sec.ResourceRef.Name == config.SecretName {
+			return false
+		}
+	} else {
+		shoot.Spec.Resources = append(shoot.Spec.Resources, gardener_types.NamedResourceReference{})
+		sec = &shoot.Spec.Resources[len(shoot.Spec.Resources)-1]
+	}
+
+	changed = true
+
+	sec.Name = "auditlog-credentials"
+	sec.ResourceRef.APIVersion = "v1"
+	sec.ResourceRef.Kind = "Secret"
+	sec.ResourceRef.Name = config.SecretName
```

Comment:
It would change the behaviour. There are 3 branches in the `if`:
- secret exists
  - and is configured correctly (`return`)
  - and is not configured correctly
- secret doesn't exist (create it)